include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -3111,7 +3158,31 @@ MGTwoLevelTransfer<dim, LinearAlgebra::distributed::Vector<Number>>::
   else if (fine_element_is_continuous)
     vec_fine_ptr->zero_out_ghost_values(); // external vector
 
-  dst.copy_locally_owned_data_from(this->vec_coarse);
+  if (use_dst_inplace == false)
+    dst.copy_locally_owned_data_from(this->vec_coarse);
+}
+
+
+
+template <int dim, typename Number>
+void
+MGTwoLevelTransfer<dim, LinearAlgebra::distributed::Vector<Number>>::
+  enable_inplace_operations_if_possible(
+    const std::shared_ptr<const Utilities::MPI::Partitioner>
+      &partitioner_coarse,
+    const std::shared_ptr<const Utilities::MPI::Partitioner> &partitioner_fine)
+{
+  if (this->partitioner_coarse->is_globally_compatible(*partitioner_coarse))
```

Comment:
This is still a bit conservative but I'll add a better heuristic (as can be found in `MGTransferMatrixFree`) in a follow-up PR, since it also requires the renumbering of the local indices.